gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1317,6 +1332,18 @@ gst_tensor_src_iio_start (GstBaseSrc * src)
         (gulong) self->sampling_frequency);
     goto error_trigger_free;
   } else {
+    /** store the default frequency */
+    filename =
+        g_build_filename (self->device.base_dir, "sampling_frequency", NULL);
+    if (!g_file_get_contents (filename, &file_contents, &length, NULL)) {
+      GST_WARNING_OBJECT (self, "Unable to read default sampling frequency.");
+    } else if (file_contents != NULL) {
+      self->default_sampling_frequency =
+          g_ascii_strtoull (file_contents, NULL, 10);
+    }
+    g_free (file_contents);
+    g_free (filename);
```

Comment:
This function is getting too long & complex. Can you refactor it later?

Reply:
Yes, opened issue #1300